src/Controls/src/Core/Application/Application.cs

```diff
@@ -99,17 +99,20 @@ public IAppLinks AppLinks
 
 				OnPropertyChanging();
 
+				if (MainPage is not null)
+				{
+					((IElementDefinition)this).RemoveResourcesChangedListener(MainPage.OnParentResourcesChanged);
+				}
+
 				_singleWindowMainPage = value;
```

Comment:
this is actually setting MainPage, so listener was never removed on the previous MainPage